src/earn/poolInfo.ts

```diff
@@ -32,3 +36,47 @@ export async function fetchAavePoolInfo(assetAddress: Address) {
     throw err
   }
 }
+
+/** @beta - Exclude from Knip dep check */
+export async function fetchAavePoolUserBalance({
+  assetAddress,
+  walletAddress,
+}: {
+  assetAddress: Address
+  walletAddress: Address
+}) {
+  try {
+    if (!assetAddress) {
+      throw new Error('No asset address provided')
+    }
+    if (!walletAddress) {
+      throw new Error('No wallet address provided')
+    }
+    Logger.debug(TAG, 'Fetching Aave pool user balance', { assetAddress, walletAddress })
+    const result = await publicClient[Network.Arbitrum].readContract({
```

Comment:
Dumb q: is this the approach aave recommends instead of saving the ERC-20 address (or hard coding it)?

Reply:
This was the approach used in `fetchAavePoolInfo` and the example shared by @jh2oman in the example code I am not sure why it is preferred.

Reply:
the downside is that, possibly, the pool contract would return a different address at some point :shrug: 

we already save the aTokenAddress is address-metadata too, so it's unexpected to fetch it here, but in some parts of the experience rely on the address in address-metadata.

Reply:
@silasbw It looks like aTokenAddress resolves to `0x724dc807b04555b71ed48a6896b6f41593b8c637` on mainnet which is  [here](https://github.com/valora-inc/address-metadata/blob/main/src/data/mainnet/arbitrum-one-tokens-info.json#L43) in address-metadata. Are you proposing we filter tokens using a hardcoded tokenId or symbol?

Reply:
I'm proposing we have one source of truth for the `aTokenAddress`, yeah.  Filtering by tokenId could work.  How are we going to associate the aave-specific logic with the aArbUSDCn token details screen?

Reply:
At this point I am thinking that a network config variable is probably simplest either `aArbUSDCnTokenId` or `aArbUSDCnAddress`.

Reply:
Added the tokenId to the networkConfig in 9873c6b7f022df5966679eb368d6fb222a6a7f0d and adjusted the fetchAavePoolUserBalance in f79cee8371a1922e4d5d08ff7103aba66f86b4cb.